app/forms/sections/items/ItemPackagingQuantityFormProvider.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.sections.items
+
+import forms.mappings.Mappings
+import models.GoodsTypeModel.GoodsType
+import play.api.data.Form
+import play.api.i18n.Messages
+
+import javax.inject.Inject
+
+class ItemPackagingQuantityFormProvider @Inject() extends Mappings {
+
+  import ItemPackagingQuantityFormProvider._
+
+  def apply(goodsType: GoodsType)(implicit messages: Messages): Form[BigDecimal] =
+    Form(
+      fieldName -> text(requiredErrorKey(goodsType))
+        .verifying(isDecimal(invalidCharactersErrorKey))
+        .verifying(maxLength(maxDigits, maxLengthErrorKey))
+        .transform[BigDecimal](BigDecimal(_), _.toString())
+        .verifying(maxDecimalPlaces(maxDecimalPlacesValue, decimalPlacesErrorKey))
+    )
+}
```

Comment:
As Item quantity is always a whole number this Form would be better as an `Int` - I don't think a `BigDecimal` makes sense.